### ISSUE TITLE
Error training with detectNet

### ISSUE THREAD

**Issue by douzsh** [NVIDIA/DIGITS#864]
I noticed your members are working on detectnet now.
On 0.15 version detectnet prototxt there is a param like this to deal with rect groups.
```
python_param {
module: 'caffe.layers.detectnet.clustering'
layer: 'ClusterDetections'
param_str : '1248, 352, 16, 0.6, 3, 0.02, 22'
}
```
In python/caffe/layers/detectnet/clustering.py comment like this:
```
# gridbox_cvg_threshold,gridbox_rect_threshold,gridbox_rect_eps,min_height
param_str : '1248,352,16,0.05,1,0.025,22'
```

Which param is better? 
In fact, I trained model with your detection walkthrough branch on NVIDIA/DIGITS.
Everything is fine but when I want to use the caffemodel, there's no rect in result.
After I changed the coverage threshold from 0.6 to 0.05, I can get lots of rects...
Is this a bug or not?
Can you help me with this?

**Comment by lukeyeager** [comment]
I saw you correctly moved this out of BVLC/caffe (https://github.com/BVLC/caffe/issues/4336). Technically the best place to ask this would be at NVIDIA/caffe, but I'll address it here.

I'm not an expert in the technical details of this model, but it makes intuitive sense to me that if you decrease the threshold then you'll get more bounding boxes to show up. Doesn't sounds like a bug to me.

I'd go with the defaults in `detectnet_network.prototxt` rather than the defaults in the comments. Did you carefully follow [all these steps](https://github.com/NVIDIA/DIGITS/blob/17adeab4ac/examples/object-detection/README.md#detectnet) to set up your hyperparameters correctly? Which dataset are you using?

**Comment by douzsh** [comment]
@lukeyeager Thanks for your quickly reply. I trained the model with kitti data same as you. The different is I didnot split the training data into 2 parts. I used them both in training and validation. I don't think this will lead to any problem.
I followed your steps in object detection walkthrough examples carefully. I also checked the lmdb data generated from KITTI. There's no problem there. Pretrained googlenet is also same as you mentioned.
Thatz very strange I did not get a correct detection. I will show you the detail of training below.
![mmexport1466558668686](https://cloud.githubusercontent.com/assets/6712525/16251763/6b22823c-37ee-11e6-8bbf-ad07495a5228.jpg)
![mmexport1466558674806](https://cloud.githubusercontent.com/assets/6712525/16251765/747e0ee6-37ee-11e6-8970-8c96b3add5e7.jpg)

**Comment by gheinrich** [comment]
Can you show the batch size graph too? Also if you can zoom in on the loss graph (using the smaller sub graph, select a narrower x-axis range e.g. epochs 5--10) so we can see more clearly how the loss functions are moving.

**Comment by IsaacYangSLA** [comment]
@douzsh , Can you also double-check your solver type and learning rate?  To me, it seemed like the solver ran away from the pre-trained and began with random guess.

**Comment by rodrigoberriel** [comment]
Is there any technical paper on DetectNet? I successfully trained using KITTI dataset (thanks to the documentation), but using another dataset, I could not get `mAP > 0`, even though the losses are continuously (slowly) decreasing. Does anyone know what could be leading to this behavior?

Btw, @douzsh, I just followed their documentation and it worked just fine, but I did the split (20% for validation).

Another dataset:
![image](https://cloud.githubusercontent.com/assets/7027770/16603065/70b22aca-42ef-11e6-82ef-7160e552064b.png)

**Comment by douzsh** [comment]
@IsaacYangSLA Thanks very much. I checked my solver again. I set the base_lr as default 0.001 but not 0.0001. I think it really ran away...

**Comment by gheinrich** [comment]
@rodrigoberriel there is more documentation coming (bear in mind that Object Detection is still a "preview" feature in DIGITS until DIGITS 4.0 gets released). You might want to check the size of your bounding boxes. The size of the receptive fields in DetectNet is 555x555 pixels and DetectNet works best for objects that are between 50 and 400 pixels in size.

**Comment by rodrigoberriel** [comment]
@gheinrich Thank you for the info. I am looking forward for the documentation.

**Comment by IsaacYangSLA** [comment]
@douzsh, good to hear that.  I found it helped set tiny learning rate when loading optimal pre-trained weights.  If you can find solver.prototxt for that specific weight, try to manually calculate its final learning rate.  It will be a good starting point to reuse pre-trained weights.
